mmv1/third_party/terraform/services/apphub/data_source_apphub_discovered_workloads.go

```diff
@@ -0,0 +1,177 @@
+                                                },
+                                                "workload_properties": {
+                                                        Type:     schema.TypeList,
+                                                        Computed: true,
+                                                        Elem: &schema.Resource{
+                                                            Schema: map[string]*schema.Schema{
+                                                                "gcp_project":{
+                                                                    Type: schema.TypeString,
+                                                                    Computed: true,
+                                                                },
+                                                                "location":{
+                                                                    Type: schema.TypeString,
+                                                                    Computed: true,
+                                                                },
+                                                                "zone":{
+                                                                    Type: schema.TypeString,
+                                                                    Computed: true,
+                                                                },
+                                                            },
+                                                        },        
+                                                },
+                                        },
+                                },
+                        },
+        },
+    }
+}
+
+func dataSourceApphubDiscoveredWorkloadRead(d *schema.ResourceData, meta interface{}) error {
+        config := meta.(*transport_tpg.Config)
+        userAgent, err := tpgresource.GenerateUserAgentString(d, config.UserAgent)
+        if err != nil {
+                return err
+        }
+
+        url, err := tpgresource.ReplaceVars(d, config, "{{ApphubBasePath}}projects/{{project}}/locations/{{location}}/discoveredWorkloads:lookup?uri={{workload_uri}}" )
+        if err != nil {
+                return err
+        }
+
+        billingProject := ""
+
+        // err == nil indicates that the billing_project value was found
+        if bp, err := tpgresource.GetBillingProject(d, config); err == nil {
+                billingProject = bp
+        }
+
+        res, err := transport_tpg.SendRequest(transport_tpg.SendRequestOptions{
+                Config:    config,
+                Method:    "GET",
+                Project:   billingProject,
+                RawURL:    url,
+                UserAgent: userAgent,
+        })
+        fmt.Println(res, err, "Test print")
+        if err != nil {
+                return transport_tpg.HandleNotFoundError(err, d, fmt.Sprintf("ApphubDiscoveredWorkload %q", d.Id()))
```

Comment:
`HandleDataSourceNotFoundError` is better.

Reply:
Done.